pyuvdata/uvdata.py

```diff
@@ -588,28 +582,28 @@ def unphase_to_drift(self):
         phs = np.exp(-1j * 2 * np.pi * (-1) * w_lambda[:, None, :, None])
         self.data_array *= phs
 
-        unique_times = np.unique(self.time_array)
-        for jd in unique_times:
+        unique_times, unique_inds = np.unique(self.time_array, return_index=True)
+        for ind, jd in enumerate(unique_times):
             inds = np.where(self.time_array == jd)[0]
+            lst = self.lst_array[unique_inds[ind]]
             obs.date, obs.epoch = self.juldate2ephem(
                 jd), self.juldate2ephem(jd)
             phase_center.compute(obs)
             phase_center_ra, phase_center_dec = phase_center.a_ra, phase_center.a_dec
-            zenith_ra = obs.sidereal_time()
-            zenith_dec = latitude
-            self.zenith_ra[inds] = zenith_ra
-            self.zenith_dec[inds] = zenith_dec
 
             # generate rotation matrices
             m0 = uvutils.top2eq_m(0., phase_center_dec)
-            m1 = uvutils.eq2top_m(phase_center_ra - zenith_ra, zenith_dec)
+            m1 = uvutils.eq2top_m(phase_center_ra - lst, latitude)
```

Comment:
There were always very small differences in our phasing related to whether I used things like zenith_ra or lst here.  I think we asserted some of that stemmed from pyephem vs astropy calculating some quantities.  Maybe it's now more consistent?  I think this deserves a deeper investigation (a la @adampbeardsley) to see how our phasing errors might change with this code change.  If I recall, we lowered our tolerances to gloss over phasing errors this small.